src/audio/dcblock/dcblock.c

```diff
@@ -0,0 +1,430 @@
+	buffer_lock(sourceb, &flags);
+
+	/* Get source, sink, number of frames etc. to process. */
+	comp_get_copy_limits(sourceb, sinkb, &cl);
+
+	buffer_unlock(sinkb, flags);
+	buffer_unlock(sourceb, flags);
+
+	dcblock_process(dev, sourceb, sinkb,
+			cl.frames, cl.source_bytes, cl.sink_bytes);
+
+	return 0;
+}
+
+/**
+ * \brief Prepares DC Blocking Filter component for processing.
+ * \param[in,out] dev DC Blocking Filter base component device.
+ * \return Error code.
+ */
+static int dcblock_prepare(struct comp_dev *dev)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+	struct sof_ipc_comp_config *config = dev_comp_config(dev);
+	struct comp_buffer *sourceb;
+	struct comp_buffer *sinkb;
+	uint32_t sink_period_bytes;
+	int ret;
+
+	comp_info(dev, "dcblock_prepare()");
+
+	ret = comp_set_state(dev, COMP_TRIGGER_PREPARE);
+	if (ret < 0)
+		return ret;
+
+	if (ret == COMP_STATUS_STATE_ALREADY_SET)
+		return PPL_STATUS_PATH_STOP;
+
+	/* DC Filter component will only ever have one source and sink buffer */
+	sourceb = list_first_item(&dev->bsource_list,
+				  struct comp_buffer, sink_list);
+	sinkb = list_first_item(&dev->bsink_list,
+				struct comp_buffer, source_list);
+
+	/* get source data format */
+	cd->source_format = sourceb->stream.frame_fmt;
+
+	/* get sink data format and period bytes */
+	cd->sink_format = sinkb->stream.frame_fmt;
+	sink_period_bytes =
+			audio_stream_period_bytes(&sinkb->stream, dev->frames);
+
+	if (sinkb->stream.size < config->periods_sink * sink_period_bytes) {
+		comp_err(dev, "dcblock_prepare(), sink buffer size %d is insufficient",
+			 sinkb->stream.size);
+		ret = -ENOMEM;
+		goto err;
+	}
+
+	dcblock_init_state(cd);
+
+	cd->dcblock_func = dcblock_find_func(cd->source_format);
+	if (!cd->dcblock_func) {
+		comp_err(dev, "dcblock_prepare(), No processing function matching frames format");
+		ret = -EINVAL;
+		goto err;
+	}
+
+	comp_info(dev, "dcblock_prepare(), source_format=%d, sink_format=%d",
+		  cd->source_format, cd->sink_format);
+
+	return 0;
+
+err:
+	comp_set_state(dev, COMP_TRIGGER_RESET);
+	return ret;
+}
+
+/**
+ * \brief Resets DC Blocking Filter component.
+ * \param[in,out] dev DC Blocking Filter base component device.
+ * \return Error code.
+ */
+static int dcblock_reset(struct comp_dev *dev)
+{
+	comp_info(dev, "dcblock_reset()");
+
+	comp_set_state(dev, COMP_TRIGGER_RESET);
+	return 0;
+}
+
+/** \brief DC Blocking Filter component definition. */
+static const struct comp_driver comp_dcblock = {
+	.type = SOF_COMP_DCBLOCK,
+	.uid  = SOF_UUID(dcblock_uuid),
+	.ops  = {
+		 .new		= dcblock_new,
+		 .free		= dcblock_free,
+		 .params	= dcblock_params,
+		 .cmd		= dcblock_cmd,
+		 .trigger	= dcblock_trigger,
+		 .copy		= dcblock_copy,
+		 .prepare	= dcblock_prepare,
+		 .reset		= dcblock_reset,
+	},
+};
+
+static SHARED_DATA struct comp_driver_info comp_dcblock_info = {
+	.drv = &comp_dcblock,
+};
+
+static void sys_comp_dcblock_init(void)
+{
+	comp_register(&comp_dcblock_info);
```

Comment:
Please change to:
`comp_register(platform_shared_get(&comp_dcblock_info, sizeof(comp_dcblock_info)));`

Reply:
@tlauda looks like we should do the ```platform_shared_get()``` inside ```comp_register()``` and perhaps for other comp calls ?

Reply:
ah i didn't even have that in my local repo. Thanks @tlauda